### PR TITLE
Port fp64 module to UBO

### PR DESCRIPTION
**Pull request by felixpalmer**
For https://github.com/visgl/deck.gl/issues/8997, once merged will also apply to `9.1-release` to unblock https://github.com/visgl/deck.gl/pull/9160

#### Change List
- Copy across webgl1 implementation
- Add UBO block
- Use `defaultUniforms` instead of `getUniforms`
- Types
- Re-instate tests

### CODE REVIEW

File: modules/shadertools/src/index.ts [hunk 1]
```diff
@@ -93,7 +93,7 @@ export {pbrMaterial} from './modules/lighting/pbr-material/pbr-material';
 // DEPRECATED - v8 legacy shader modules (non-uniform buffer)
 
 // math libraries
-export {fp64, fp64arithmetic} from './modules-webgl1/math/fp64/fp64';
+// export {fp64, fp64arithmetic} from './modules-webgl1/math/fp64/fp64';
```
Comment (ibgreen) [inline modules/shadertools/src/index.ts:96]: Can we cut instead? At least on master which is targeting 9.2

**Review by ibgreen — APPROVED** [review]
Big thanks for this @felixpalmer

File: modules/shadertools/src/index.ts (hunk 1 above)
Comment (felixpalmer) [reply modules/shadertools/src/index.ts:96]: I'm in favour of removing all the old `modules-webgl1` code & exports. I think it has all been ported

**Comment by felixpalmer** [comment]
@ibgreen FYI when cherry-picking into 9.1 I removed:

 `import './modules-webgl1/fp64/fp64-arithmetic-transform.spec';`

from `modules/shadertools/test/index.ts` to match 9.2

**Pull request merged**